### PR TITLE
Added route meta field and navigation guards to /search route  

### LINKED ISSUE #853: search route without any query params is showing a blank search result page
## Describe the bug
If we don't pass any query params to the `/search` route it displays a black page

## To Reproduce
1. Go to 'https://ccsearch.creativecommons.org/search or got to just `search` route locally'

## Expected behavior
Should fall back to the `page not found` page

## Screenshots
search_blank

## Desktop
 - OS: Ubuntu 19.10
 - Browser Chrome
 - Version 80.0.3987.162

## Additional context
I want to work on this issue :)

### PR DESCRIPTION
**Pull request by abhisheknaiidu**
## Fixes
Fixes #853 

## Description
Added route meta field and navigation guards to /search route  
Added with the help of https://router.vuejs.org/guide/advanced/meta.html 

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/router/client.js [hunk 1]
```diff
@@ -80,6 +84,26 @@ const router = new VueRouter({
   },
 });
 
+// eslint-disable-next-line
+router.beforeEach((to, from, next) => {
```
Comment (brenoferreira) [inline src/router/client.js:88]: the redirect should happen on the server, not the on client.

File: src/router/client.js [hunk 2]
```diff
@@ -80,6 +84,26 @@ const router = new VueRouter({
   },
 });
 
+// eslint-disable-next-line
```
Comment (brenoferreira) [inline src/router/client.js:87]: Disabling ESLint is discouraged. Please remove this

File: src/router/client.js (hunk 1 above)
Comment (abhisheknaiidu) [reply src/router/client.js:88]: @brenoferreira first I changed in `router/server.js` but its not working in there so, I then changed it in client-side, Should I need to change the 'serverEntry` to get it worked?

**Comment by abhisheknaiidu** [comment]
@brenoferreira Yess, now i added the meta field and navigation guard to `server` file, Could you once review it? but its not working in `localhost:8443` (Client Side)

File: src/router/client.js (hunk 2 above)
Comment (abhisheknaiidu) [reply src/router/client.js]: Done

File: src/router/client.js (hunk 1 above)
Comment (brenoferreira) [reply src/router/client.js]: Yes. Server is accessible via port 8080

Comment (abhisheknaiidu) [reply src/router/client.js]: Yess, I added it to `server`

**Pull request merged**